userspace/falco/app_actions/configure_interesting_sets.cpp

```diff
@@ -52,10 +131,113 @@ void application::configure_interesting_sets()
 		}
 	}
 
-	/* In this case we get the tracepoints for the `libsinsp` state and we remove
-	 * the `sched_switch` tracepoint since it is highly noisy and not so useful
+	/* Reading a scap file we have no concepts of ignored events we read all we need. */
+	if(!m_options.all_events && !is_capture_mode())
+	{
+		/* Here we have already initialized the application state with the interesting syscalls,
+		 * so we have to check if any event types used by the loaded rules are not considered by
+		 * Falco interesting set.
+		 */
+		check_for_ignored_events();
+	}
+
+}
+
+void application::activate_interesting_syscalls(std::unique_ptr<sinsp>& inspector, const std::unordered_set<std::string>& rules_evttypes_names)
+{
+
+	/* Translate PPME event names to PPM syscall idx codes.
+	 * PPM syscall idx codes can be viewed as condensed libsinsp lookup table to map a system call name to it's actual system syscall id (as defined by the Linux kernel).
+	 * Hence here we don't need syscall enter and exit distinction.
+	*/
+	std::unordered_set<uint32_t> rules_ppm_sc_set = get_syscalls_ppm_codes(rules_evttypes_names);
+	std::unordered_set<std::string> rules_syscalls_names = inspector->get_syscalls_names(rules_ppm_sc_set);
+	if (rules_syscalls_names.size() > 0)
+	{
+		falco_logger::log(LOG_INFO, "(" + std::to_string(rules_syscalls_names.size()) + ") syscalls activated in rules: " + concat_syscalls_names(rules_syscalls_names) + "\n");
+	}
+
+	/*
+	*
+	* DEFAULT OPTION:
+	*
+	* Current enforce_simple_ppm_sc_set approach includes multiple steps:
+	* (1) Enforce all positive syscalls from each Falco rule
+	* (2) Enforce a static set of syscalls in addition to the syscalls defined in Falco's rules
+	* (3) Enforce `libsinsp` state set (non-adaptive, not conditioned by rules, but based on PPME event table flags indicating generic sinsp state modifications)
+	* -> Final set is union of (1), (2) and (3)
+	*
+	*/
+
+	/* Derive union of rules_ppm_sc_set (all syscalls defined in Falco rules) and enforced syscalls for libsinsp state and declare ppm_sc_of_interest. */
+	m_state->ppm_sc_of_interest = inspector->enforce_simple_ppm_sc_set(rules_ppm_sc_set);
```

Comment:
Reminder for the futures us: once we move from the simple set to the sinsp state minimal set, change this to `enforce_sinsp_state_ppm_sc` and update the comment above.